modules/integration/tests-common/common-artifacts/carbon-context/src/main/java/org/wso2/appserver/integration/common/artifact/carboncontext/TenantServlet.java

```diff
@@ -0,0 +1,193 @@
+                break;
+            case UNLOAD_TENANT:
+                unloadTenant(request, response);
+                break;
+            case ACCESS_REGISTRY_WITHOUTOUT_LOAD:
+                accessRegistryWithoutLoad(request, response);
+                break;
+        }
+    }
+
+    private void accessRegistryWithoutLoad(HttpServletRequest request, HttpServletResponse response) {
+        String tenantId = request.getParameter("tenantId");
+        if (tenantId != null && !tenantId.isEmpty()) {
+            try {
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(Integer.parseInt(tenantId), true);
+                Registry registry = PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                                                           .getRegistry(RegistryType.USER_CONFIGURATION);
+                Resource resource = registry.newResource();
+                resource.setContent("TestResource".getBytes());
+                registry.put("TestResourcePath", resource);
+            } catch (RegistryException e) {
+                if (log.isDebugEnabled()) {
+                    log.error(e.getMessage(), e);
+                }
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        } else {
+            log.info("tenantId should be initialized");
+        }
+    }
+
+    private void unloadTenant(HttpServletRequest request, HttpServletResponse response) {
+        String tenantId = request.getParameter("tenantId");
+        if (tenantId != null && !tenantId.isEmpty()) {
+            PrivilegedCarbonContext.unloadTenant(Integer.parseInt(tenantId));
+        } else {
+            log.info("tenantId should be initialized");
+        }
+    }
+
+    private void setApplicationName(HttpServletRequest request, HttpServletResponse response) {
+        String appName = request.getParameter("appName");
+        if (appName != null && !appName.isEmpty()) {
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setApplicationName(appName);
+            response.addHeader("set-app-name",
+                               PrivilegedCarbonContext.getThreadLocalCarbonContext().getApplicationName());
+        } else {
+            log.info("appName need to be initialized");
+        }
+    }
+
+    private void setUsername(HttpServletRequest request, HttpServletResponse response) {
+        String username = request.getParameter("username");
+        if (username != null && !username.isEmpty()) {
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(username);
+            response.addHeader("set-username", PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername());
+        } else {
+            log.info("username should be initialized");
+        }
+    }
+
+    private void getRegistry(HttpServletRequest request, HttpServletResponse response) {
```

Comment:
Any reason for this method to not return any , even-though the name is getRegistry?